src/main/scala/projectEuler/LargestPrimeFactorCalculator.scala

```diff
@@ -0,0 +1,26 @@
+package projectEuler
+
+import scala.math._
+
+/**
+ * Problem 3. Largest prime factor
+ * The prime factors of 13195 are 5, 7, 13 and 29.
+ * What is the largest prime factor of the number 600851475143 ?
+ * @see https://projecteuler.net/problem=3
+ */
+class LargestPrimeFactorCalculator(num: Long) {
+
+  private def isFactor(numerator: Long)(denominator: Long): Boolean = {
+    numerator % denominator == 0
+  }
+
+  private def isPrime(number: Long): Boolean = {
+    !(2L until number exists isFactor(number))
```

Comment:
Instead of `isFactor` being a curried function, use the placeholder `_` to create a single-parameter predicate, i.e. `!(2L until number exists isFactor(number, _))`